docs/NFF-0.1-spec.norg

```diff
@@ -0,0 +1,500 @@
+
+**** Punctuation
+     A {> character} is considered to be *punctuation* if it is any of the following:
+     - a standard ASCII punctuation character: |`!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~`|
+     - any punctuation character specific to the current locale
+       +TODO(Vhyrro): check if this is what we really want. Currently taken from std::iswpunct docs+
+
+**** Escaping
+     A single {> character} can be escaped if it is immediately preceded by a backslash, `\`
+     (`U+005C`).
+
+**** Special Character
+     - | | TODO: formalize this section
+     - Emojis
+     - Should we support [entity references]?
+
+     [entity references]{https://spec.commonmark.org/0.30/#entity-references}
+
+*** Words
+    The _Norg_ format is designed to be parsed on a word-by-word basis from left-to-right through
+    the entire document /in a single pass/. This is possible, because the language is [Free-form],
+    meaning that whitespace has no semantic meaning, and because the markup follows strict rules
+    which are outlined in the later sections of this document.
+
+    A *word* is considered any combinations of {> character}s which are neither {# Whitespace} nor
+    {# Punctuation} or other {# Special Character}.
+
+*** Paragraph Segments
+    {# Words} are first combined into {# paragraph segments}. A segment gets terminated by any of
+    the following:
+    - a {> line ending}
+    - any of the {# attached modifiers}
+    - any of the {# linkables}
+
+*** Paragraphs
+    Paragraphs are then formed of consecutive {# paragraph segments}, {# attached modifiers} or
+    {# linkables}. A paragraph is terminated by:
+    - a _paragraph break_ (which equals two consecutive {> line endings})
+    - any of the {# detached modifiers}
+    - any of the {# delimiting modifiers}
+
+** Detached Modifiers
+   _Norg_ has multiple detached modifiers. The name originates from their differentiation to the
+   {# Attached Modifiers} which will be discussed later.
+   All detached modifiers must abide by the following rules:
+   - a detached modifier can _only_ occur at the beginning of the line (an arbitrary {# whitespace}
+     (but nothing else!) may precede it)
+   - a detached modifier must be immediately followed by {# whitespace} (incl. a {> line ending}) or
+     another detached modifier of the same type
+
+   The following list are the valid *detached* modifiers. The list also adds three possible
+   properties to each category which will be explained in more detail below.
+   - `*`: {# Headings} (structural, nestable)
+   - `%`: {# Markers} (structural)
+   - `-`: {# Unordered Lists} (nestable)
+   - `~`: {# Ordered Lists} (nestable)
+   - `>`: {# Quotes} (nestable)
+   - `$`: {# Definitions} (range-able)
+   - `^`: {# Footnotes} (range-able)
+   - `=`: {# Insertions} (range-able)
+   - | | `?`: {# Multiple Choice Questions} +TODO: decide whether or not to make these range-able+
```

Comment:
I'm not sure on this one. The `?` detached modifier will be a bit special already because it will have restricted contents.
Since we now have the `\ ... ---` syntax for allowing e.g. code blocks within detached modifier content this could be used to insert more information into the question title. Thus, I don't think range-ability makes sense here but I wanted to discuss this before fixing this.